editor/src/clj/internal/graph/types.clj

```diff
@@ -62,11 +63,13 @@
   (.write writer (str (.-label ep)))
   (.write writer "]"))
 
-(defn- read-endpoint [[node-id-expr label-expr]]
-  `(->Endpoint ~node-id-expr ~label-expr))
+(defonce ^WeakInterner endpoint-interner (WeakInterner. 65536))
```

Comment:
`65536` is sufficient for small projects like our examples, but for the large project that prompted this investigation, this will double several times until it reaches `16777216`. We spend about half a second in total rehashing and copying the internal hash map during this process.